runtime/src/bank.rs

```diff
@@ -1637,7 +1644,13 @@ impl Bank {
         // (total_rewards, distributed_rewards, credit_end_exclusive), total capital will increase by (total_rewards - distributed_rewards)
         self.create_epoch_rewards_sysvar(total_rewards, distributed_rewards, credit_end_exclusive);
 
-        self.create_epoch_rewards_partition_data_account(num_partitions, parent_blockhash);
+        // Skip creating data account when we are testing partitioned rewards
+        let should_create = self.is_partitioned_rewards_code_enabled()
+            || !self.force_partition_rewards_in_first_block_of_epoch();
```

Comment:
I think this should be an `&&` not an `||`.

In fact, I think `self.is_partitioned_rewards_code_enabled()` will always evaluate to true (and therefore should be removed), because we only reach `begin_partitioned_rewards()` in that case: https://github.com/solana-labs/solana/blob/6c4746dc14cddd4161e684048e46f3c367da45c5/runtime/src/bank.rs#L1543-L1545

It would probably be more clear if L1543-L1545 used `Bank::is_partitioned_rewards_code_enabled()` instead of duplicating that code.

Reply:
Fixed.